### PR TITLE
[Infra] Verifying the main project repo based on a git ref

### PR DESCRIPTION
**Pull request by Leo-Neat**

### CODE REVIEW

File: infra/base-images/base-builder/detect_repo.py [hunk 1]
```diff
@@ -63,15 +71,33 @@ def get_repo(repo_path):
...
+    ref: The git ref that is being checked for
+
+  """
+  # Check if valid git repo.
+  if not os.path.exists(os.path.join(repo_path, '.git')):
+    return False
+
+  execute(['git', 'fetch', '--all'], location=repo_path)
+  _, return_code = execute(['git', 'rev-parse', ref], location=repo_path)
+  if return_code != 0:
```
Comment (Dor1s) [inline infra/base-images/base-builder/detect_repo.py]: can simplify to `return return_code == 0`

File: infra/base-images/base-builder/detect_repo_test.py [hunk 2]
```diff
@@ -62,7 +62,32 @@ def test_infer_main_repo(self):
...
+    """Tests that the main repo can be inferred based on an example commit."""
+
+    with tempfile.TemporaryDirectory() as tmp_dir:
+
+      # Construct example repo's to check for commits.
+      repo_manager.RepoManager('https://github.com/curl/curl.git', tmp_dir)
+      self.check_commit_with_repo(
+          'https://github.com/curl/curl.git',
+          'curl',
+          'ee5f192c325e9063e2b4d32eff1d393b75a42f0b',
```
Comment (Dor1s) [inline infra/base-images/base-builder/detect_repo_test.py]: why does this pass both `commit` and `ref` values?

File: infra/base-images/base-builder/detect_repo_test.py [hunk 3]
```diff
@@ -75,12 +100,17 @@ def check_commit_with_repo(self, repo_origin, repo_name, commit, tmp_dir):
         'python3', 'detect_repo.py', '--src_dir', tmp_dir, '--example_commit',
```
Comment (Dor1s) [inline infra/base-images/base-builder/detect_repo_test.py]: don't pass commit if it's not specified, as you need to remove it from the new test, if I understand correctly

File: infra/base-images/base-builder/detect_repo_test.py [hunk 4]
```diff
@@ -75,12 +100,17 @@ def check_commit_with_repo(self, repo_origin, repo_name, commit, tmp_dir):
...
     ]
+    if ref:
+      command.extend(['--ref', ref])
     out, _ = detect_repo.execute(
         command, location=os.path.dirname(os.path.realpath(__file__)))
     match = re.search(r'\bDetected repo: ([^ ]+) ([^ ]+)', out.rstrip())
     if match and match.group(1) and match.group(2):
       self.assertEqual(match.group(1), repo_origin)
       self.assertEqual(match.group(2), repo_name)
+    else:
```
Comment (Dor1s) [inline infra/base-images/base-builder/detect_repo_test.py]: what is this part for? why would anyone call this function with `None` repo origin and repo name?

**Comment by jonathanmetzman** [comment]
Was this intentionally closed?

**Comment by Leo-Neat** [comment]
> Was this intentionally closed?

Yes, I am working on a better edition.

**Pull request closed without merging**